packages/private-apis/src/implementation.ts

```diff
@@ -168,10 +166,10 @@ function lock( object, privateData ) {
  * // { a: 1 }
  * ```
  *
- * @param {any} object The object to unlock the private data from.
- * @return {any} The private data bound to the object.
+ * @param object The object to unlock the private data from.
+ * @return The private data bound to the object.
  */
-function unlock( object ) {
+function unlock( object: Record< symbol, WeakKey > ) {
```

Comment:
A little better type would be `unlock<T>(object: unknown): T` because:
- the function can safely work with any `object` and try to unlock it. We might run into problems with the `Record` type if the locked object doesn't have a compatible type. That will probably happen a lot as we migrate the poorly-typed-yet codebase.
- we should be able to cast the return value to be an object of known type with private methods.

I predict that we'll run into these issues as we use the typed `lock`/`unlock` and will eventually come back to revisit this.

Reply:
Updated in #66682

Reply:
That signature seems to break a lot of existing poorly typed usage of `lock` and `unlock`